src/olympia/addons/tests/test_views.py

```diff
@@ -4048,6 +4048,75 @@ def test_cant_upload_source_for_reviewed_file(self):
+        assert response.status_code == 400, response.content
+        assert response.data == {'is_disabled': ['File is already disabled.']}
+
+        # But if the file isn't disabled the version should be good to disable
+        self.version.file.update(status=amo.STATUS_APPROVED)
+        response = self.client.patch(self.url, data={'is_disabled': True})
+
+        assert response.status_code == 200, response.content
+        assert response.data['is_disabled'] is True
+        self.version.reload()
+        self.version.file.reload()
+        assert self.version.is_user_disabled
+        assert self.version.file.status == amo.STATUS_DISABLED
+        alog = ActivityLog.objects.exclude(
+            action__in=(amo.LOG.LOG_IN.id, amo.LOG.LOG_IN_API_TOKEN.id)
+        ).get()
+        assert alog.user == self.user
+        assert alog.action == amo.LOG.DISABLE_VERSION.id
+
+    def test_cannot_disable_if_promoted(self):
+        self.make_addon_promoted(self.addon, RECOMMENDED, approve_version=True)
+
+        response = self.client.patch(self.url, data={'is_disabled': True})
+        assert response.status_code == 400
+        assert response.json()['is_disabled'][0].startswith(
+            'The latest approved version of this Recommended add-on cannot be deleted'
+        )
+        assert len(response.json()) == 1
+        assert not self.version.reload().is_user_disabled
+
+        # Now add another version so it's okay to disable one of them
+        self.version = version_factory(addon=self.addon, promotion_approved=True)
+        self.addon.reload()
+        self.url = reverse_ns(
+            'addon-version-detail',
+            kwargs={'addon_pk': self.addon.slug, 'pk': self.version.id},
+            api_version='v5',
+        )
+        response = self.client.patch(self.url, data={'is_disabled': True})
+        assert response.status_code == 200
+        self.version.reload()
+        self.version.file.reload()
+        assert self.version.is_user_disabled
+        assert self.version.file.status == amo.STATUS_DISABLED
+        alog = ActivityLog.objects.exclude(
+            action__in=(amo.LOG.LOG_IN.id, amo.LOG.LOG_IN_API_TOKEN.id)
+        ).get()
+        assert alog.user == self.user
+        assert alog.action == amo.LOG.DISABLE_VERSION.id
+
+    def test_enable_disabled_version(self):
+        self.version.is_user_disabled = True
+        response = self.client.patch(self.url, data={'is_disabled': False})
+        assert response.status_code == 200, response.content
+        assert response.data['is_disabled'] is False
+        self.version.reload()
+        self.version.file.reload()
+        assert not self.version.is_user_disabled
+        alog = ActivityLog.objects.exclude(
+            action__in=(amo.LOG.LOG_IN.id, amo.LOG.LOG_IN_API_TOKEN.id)
+        ).get()
+        assert alog.user == self.user
+        assert alog.action == amo.LOG.ENABLE_VERSION.id
+
```

Comment:
We should probably also have tests for when setting `is_user_disabled` does not modify the `file` status, to make sure we return an accurate `is_disabled` in that case.

Reply:
As far as I can see that's not a possibility - you can't set `is_user_disabled` when the file is already `amo.STATUS_DISABLED` but setting `is_user_disabled` to true ends up setting file status to `amo.STATUS_DISABLED`